src/solvers/flattening/boolbv_mult.cpp

```diff
@@ -68,37 +55,24 @@ bvt boolbvt::convert_mult(const exprt &expr)
   else if(expr.type().id()==ID_unsignedbv ||
           expr.type().id()==ID_signedbv)
   {
-    if(op0.type()!=expr.type())
-      throw "multiplication with mixed types";
-
+    INVARIANT(op0.type() == expr.type(), "multiplication with mixed types");
     bv_utilst::representationt rep=
       expr.type().id()==ID_signedbv?bv_utilst::representationt::SIGNED:
                                     bv_utilst::representationt::UNSIGNED;
 
     bv=convert_bv(op0);
-
-    if(bv.size()!=width)
-      throw "convert_mult: unexpected operand width";
-
+    INVARIANT(bv.size() == width, "convert_mult: unexpected operand width");
```

Comment:
DATA_INVARIANT

Reply:
I'm not sure about this one, it's more like we are checking that `convert_bv` did its work correctly.

Reply:
OK, CHECK_RETURN then.